src/ui/hooks.js

```diff
@@ -8,7 +8,7 @@ export function usePermission(permissionToCheck) {
   return useMemo(
     () => hasUserPermission(user, permissionToCheck),
     /* eslint-disable-next-line react-hooks/exhaustive-deps */
-    [permissionToCheck, ...user.roles]
+    [permissionToCheck, ...user?.roles || []]
```

Comment:
Actually this line was not good before, we can just pass `permissionToCheck` and `user` to the dependency array and not care when user is undefined (also I should have listened to ESLint here when I wrote this)

Reply:
Fixed